src/ndiaudiosrc.rs

```diff
@@ -365,6 +365,10 @@ impl ElementImpl for NdiAudioSrc {
                     controller.shutdown();
                 }
             }
+            gst::StateChange::ReadyToNull => {
+                let mut state = self.state.lock().unwrap();
+                state.receiver = None;
```

Comment:
This should already happen as part of https://github.com/teltek/gst-plugin-ndi/blob/092ad6b9f8903ba34830525993f48571ff17ade5/src/ndiaudiosrc.rs#L442-L448

Why is it not in your case?